sql/insert.go

```diff
@@ -118,26 +118,32 @@ func (p *planner) Insert(n *parser.Insert) (planNode, error) {
 
 		// Write the row columns.
 		for i, val := range values {
-			if _, ok := primaryKeyCols[cols[i].ID]; ok {
+			col := cols[i]
+
+			// Make sure the value can be written to the column before proceeding.
```

Comment:
Can we edit this to mention that we're checking for type compatibility between the datum and the column type.